components/brave_ads/core/internal/common/time/time_constraint_util.cc

```diff
@@ -14,12 +13,31 @@ bool DoesHistoryRespectRollingTimeConstraint(
     const std::vector<base::Time>& history,
     const base::TimeDelta time_constraint,
     const size_t cap) {
-  const size_t count = base::ranges::count_if(
-      history, [time_constraint](const base::Time created_at) {
-        return base::Time::Now() - created_at < time_constraint;
-      });
+  const base::Time now = base::Time::Now();
 
-  return count < cap;
+  size_t count = 0;
+  size_t remaining_cap = cap;
+
+  for (auto iter = history.crbegin(); iter != history.crend();
+       ++iter, --remaining_cap) {
```

Comment:
I didn't understand how the `base::ranges::count_if` was converted to this loop.
can `history` size be higher then `remaining_cap` value? If so, then we can end up with decrementing `remaining_cap` with zero value.

Reply:
Optimize again :-) Please review. Thanks